swap_meet/vendor.py

```diff
@@ -1,2 +1,175 @@
+from swap_meet.item import Item
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
 class Vendor:
-    pass
\ No newline at end of file
+    """creates a vendor object with an attribute called "inventory" and 2 methods: 
+    "add" that adds items to the inventory, and 
+    "remove" that removes item from the inventory."""
+
+    def __init__(self, inventory=None): 
+        # Note that inventory is not set to [] as default because 
+        # only immutable types should be set as default param.
+        #else each instance will share the same value of param.
+        """constructor"""
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+        
+    def add(self, item):
+        """adds given item to the inventory. Returns the item that was added."""
+
+        if isinstance(item, Item):
+            self.inventory.append(item)
+            return item
+        else:
+            raise ValueError(f"item argument passed must be of Item type")
+
+    def remove(self, item):
+        """removes given item from the inventory. Returns the item that was removed.
+        If the inventory doesn't contain that item, returns False"""
+
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError: 
+            return False
+
+    def get_by_category(self, category):
+        """returns the item that matches given category."""
+        output_list = []
+        for item in self.inventory:
+            if item.category.upper() == category.upper():  #handles edge case of incorrect casing in input
+                output_list.append(item)
+        return output_list
+    
+    def swap_items(self, friend, my_item, their_item):
+        """swaps one item between the current vendor object and their friends"""
+        if my_item in self.inventory and their_item in friend.inventory:
```

Comment:
This approach works, but consider how a `try-except` block could improve your time complexity. 🤔 💭 😌